Rules/CommonScripts/GameStateBanners.as

```diff
@@ -0,0 +1,194 @@
+#define CLIENT_ONLY
+
+Vec2f bannerStart;
+Vec2f bannerPos;
+Vec2f bannerDest;
+f32 frameTime = 0.0f;
+const f32 maxTime = 0.6f;
+
+bool minimap = true;
+
+const u32 winBannerDuration = 8 * getTicksASecond();
+const u32 buildBannerDuration = 5 * getTicksASecond();
+const u32 gameBannerDuration = 5 * getTicksASecond();
+
+namespace Banner
+{
+	enum State
+	{
+		none = 0,
+		build,
+		game,
+		win
+	};
+}
+
+void onInit(CRules@ this)
+{
+	if (!GUI::isFontLoaded("AveriaSerif-Bold_32"))
+	{
+		string AveriaSerif = CFileMatcher("AveriaSerif-Bold.ttf").getFirst();
+		GUI::LoadFont("AveriaSerif-Bold_32", AveriaSerif, 32, true);
+	}
+
+	minimap = this.minimap;
+	onRestart(this);
+}
+
+void onRestart(CRules@ this)
+{
+	ResetBannerInfo(this);
+	this.minimap = minimap;
+
+	SetBanner(this);
+}
+
+void onNewPlayerJoin(CRules@ this, CPlayer@ player)
+{
+	if (player.isMyPlayer())
+	{
+		SetBanner(this);
+	}
+}
+
+void onStateChange(CRules@ this, const u8 oldState)
+{
+	ResetBannerInfo(this);
+	SetBanner(this);
+}
+
+void onTick(CRules@ this)
+{
+	if (this.get_u8("Animate Banner") != Banner::none && this.get_u32("Banner End") < getGameTime())
+	{
+		ResetBannerInfo(this);
+	}
+}
+
+void onRender(CRules@ this)
+{
+	u8 banner_type = this.get_u8("Animate Banner");
+
+	if (banner_type != Banner::none)
+	{
+		Driver@ driver = getDriver();
+		if (driver !is null)
+		{
+			if (bannerPos != bannerDest)
+			{
+				frameTime = Maths::Min(frameTime + (getRenderDeltaTime() / maxTime), 1);
+
+				bannerPos = Vec2f_lerp(bannerStart, bannerDest, frameTime);
+			}
+
+			if (banner_type == Banner::win) 
+			{
+				DrawWinBanner(bannerPos, this.getTeamWon());
+				this.SetGlobalMessage("");
+			}
+			// todo: implement tth versions (alternatively remove TTH)
+			else if (banner_type == Banner::build && (this.gamemode_name == "CTF" || this.gamemode_name == "SmallCTF"))
+			{
+				DrawBuildBanner(bannerPos);
+			} 
+			else if (banner_type == Banner::game && (this.gamemode_name == "CTF" || this.gamemode_name == "SmallCTF"))
+			{
+				CPlayer@ p = getLocalPlayer();
+				int team = p is null ? 0 : p.getTeamNum();
+				// show flags of enemy team colour
+				if (team == 0) team = 1;
+				else team = 0;
+
+				DrawGameBanner(bannerPos, team);
+			} 
+		}
+	}
+}
+
+void ResetBannerInfo(CRules@ this)
+{
+	this.set_u8("Animate Banner", Banner::none);
+	frameTime = 0.0f;
+}
+
+void SetBanner(CRules@ this)
+{
+	Driver@ driver = getDriver();
+	if (driver !is null)
+	{
+		bannerDest = Vec2f(driver.getScreenWidth()/2, driver.getScreenHeight()/3);
+		bannerStart = bannerDest;
+		bannerStart.y = 0;
+		bannerPos = bannerStart;
+
+		if (this.isGameOver() && this.getTeamWon() >= 0)
```

Comment:
If you're using `getGameState()` for everything, it should be used here as well (`GAME_OVER`).

Additionally, store it once in a variable like `state` rather than getting it every time. You can also use `else if` rather than ifs here, as only one of those states can be active.